src/win/wgl_disp.c

```diff
@@ -959,10 +959,8 @@ static bool create_display_internals(ALLEGRO_DISPLAY_WGL *wgl_disp)
       return false;
    }
 
-   major = _al_get_suggested_display_option(disp,
```

Comment:
What is the reason for this change? Since a display is passed to the function it feels much safer using that display and not the global flags which (maybe?) could have changed in the meantime.

Reply:
_al_get_suggested_display_option is broken and does not refer to the settings requested by the user. Threading was not taken into consideration, as there are other problems with multi threaded display creation.

Reply:
al_get_new_X setting is used commonly throughout the display creation code, so it would need to be a bigger change if you wanted that

Reply:
I'm not sure _al_get_suggested_display_option is broken - but I think it contains the detected settings, not the to-be-set ones, so just the wrong function to use there. So yeah, good catch! What problems do you see with multi-threaded display creation? Either way, your patch would not affect that so looks good to me.